subprojects/language-native/src/main/java/org/gradle/language/swift/internal/DefaultSwiftBinary.java

```diff
@@ -150,4 +174,73 @@ public FileCollection getObjects() {
+        throw new UnsupportedOperationException();
+    }
+
+    private class ModulePath implements MinimalFileSet, Buildable {
+        private final Configuration importPathConfig;
+
+        private Set<File> result;
+
+        ModulePath(Configuration importPathConfig) {
+            this.importPathConfig = importPathConfig;
+        }
+
+        @Override
+        public String getDisplayName() {
+            return "Module include path for " + DefaultSwiftBinary.this.toString();
+        }
+
+        @Override
+        public Set<File> getFiles() {
+            if (result == null) {
+                result = Sets.newLinkedHashSet();
+                Map<String, ModuleMap> moduleMaps = Maps.newHashMap();
+                for (ResolvedArtifactResult artifact : importPathConfig.getIncoming().getArtifacts()) {
+                    Usage usage = artifact.getVariant().getAttributes().getAttribute(Usage.USAGE_ATTRIBUTE);
+                    if (usage != null && Usage.C_PLUS_PLUS_API.equals(usage.getName())) {
+                        String moduleName;
+
+                        ComponentIdentifier id = artifact.getId().getComponentIdentifier();
+                        if (ModuleComponentIdentifier.class.isAssignableFrom(id.getClass())) {
+                            moduleName = ((ModuleComponentIdentifier) id).getModule();
+                        } else if (ProjectComponentIdentifier.class.isAssignableFrom(id.getClass())) {
+                            moduleName = ((ProjectComponentIdentifier) id).getProjectName();
+                        } else if (LibraryBinaryIdentifier.class.isAssignableFrom(id.getClass())) {
+                            moduleName = ((LibraryBinaryIdentifier) id).getLibraryName();
+                        } else {
+                            throw new IllegalArgumentException("Could not determine the name of " + id.getDisplayName() + ": unknown component identifier type: " + id.getClass().getSimpleName());
+                        }
+
+                        ModuleMap moduleMap;
+                        if (moduleMaps.containsKey(moduleName)) {
+                            moduleMap = moduleMaps.get(moduleName);
+                        } else {
+                            moduleMap = new ModuleMap(moduleName, Lists.<String>newArrayList());
+                            moduleMaps.put(moduleName, moduleMap);
+                        }
+                        moduleMap.getPublicHeaderPaths().add(artifact.getFile().getAbsolutePath());
+                    }
+                    // TODO Change this to only add SWIFT_API artifacts and instead parse modulemaps to discover compile task inputs
+                    result.add(artifact.getFile());
+                }
+
+                if (!moduleMaps.isEmpty()) {
+                    NativeDependencyCache cache = getNativeDependencyCache();
+                    for (ModuleMap moduleMap : moduleMaps.values()) {
```

Comment:
I think this changes the ordering of the module maps based on the hash keys.  I think we'd want moduleMaps to be a linked hash map at least.

Reply:
Good catch.  Fixed.